### PR TITLE
[Pointer Events] Add a stub implementation for the `getCoalescedEvents` function of the PointerEvent interface

### PR DESCRIPTION
**Pull request by rr-codes**
#### 5da0cb3702d0173c3dff499c70c8af3e6c416a06
<pre>
[Pointer Events] Add a stub implementation for the `getCoalescedEvents` function of the PointerEvent interface
<a href="https://bugs.webkit.org/show_bug.cgi?id=276832">https://bugs.webkit.org/show_bug.cgi?id=276832</a>
<a href="https://rdar.apple.com/132105605">rdar://132105605</a>

Reviewed by Tim Nguyen.

Add a stub implementation for the `getCoalescedEvents` function of the `PointerEvent` interface,
as per the corresponding UI events specification (<a href="https://w3c.github.io/pointerevents/#dom-pointerevent-getcoalescedevents).">https://w3c.github.io/pointerevents/#dom-pointerevent-getcoalescedevents).</a>

Also slightly refactor the `PointerEvent` constructors to allow for explicitly setting the value of
`bubbles` and `isCancelable`, which must be set to `false` for the coalesced events.

Also adds a runtime feature flag to control its enablement.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/dom/PointerEvent.cpp:
(WebCore::PointerEvent::create):
(WebCore::PointerEvent::PointerEvent):
(WebCore::PointerEvent::getCoalescedEvents):
* Source/WebCore/dom/PointerEvent.h:
* Source/WebCore/dom/PointerEvent.idl:

Canonical link: <a href="https://commits.webkit.org/281173@main">https://commits.webkit.org/281173@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/8f86630044e9d561e3aba54b08c4c667a4d660f3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/59027 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/38355 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/11521 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/62702 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/9470 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/61156 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/46006 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/9675 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/47771 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6759 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/61057 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/35881 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/51072 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/28629 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/32604 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/8348 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/8474 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/52119 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/54557 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/8628 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/64401 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/58268 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/2939 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/8584 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/55093 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/2949 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/51085 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/55197 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13060 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/2469 "Passed tests") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/80029 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/34184 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/13847 "Found 12 new JSC stress test failures: stress/proxy-get-and-set-recursion-stack-overflow.js.eager-jettison-no-cjit, stress/proxy-get-and-set-recursion-stack-overflow.js.no-cjit-collect-continuously, stress/proxy-get-and-set-recursion-stack-overflow.js.no-cjit-validate-phases, stress/proxy-set.js.bytecode-cache, stress/proxy-set.js.default, stress/proxy-set.js.dfg-eager, stress/proxy-set.js.dfg-eager-no-cjit-validate, stress/proxy-set.js.eager-jettison-no-cjit, stress/proxy-set.js.mini-mode, stress/proxy-set.js.no-cjit-collect-continuously ... (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/35268 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/36353 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/35014 "Built successfully") | | |

### CODE REVIEW

File: Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml [hunk 1]
```diff
@@ -2829,6 +2829,20 @@ GenericCueAPIEnabled:
     WebCore:
       default: false
 
+GetCoalescedEventsEnabled:
+  type: bool
+  status: unstable
```
Comment (nt1m) [inline Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:2834]: Up to you really, but testable will make it run in WKTR, which we'll  want eventually.

File: Source/WebCore/dom/PointerEvent.cpp [hunk 2]
```diff
@@ -68,7 +68,12 @@ RefPtr<PointerEvent> PointerEvent::create(MouseButton button, const MouseEvent&
...
 Ref<PointerEvent> PointerEvent::create(const AtomString& type, MouseButton button, const MouseEvent& mouseEvent, PointerID pointerId, const String& pointerType)
 {
-    return adoptRef(*new PointerEvent(type, button, mouseEvent, pointerId, pointerType));
+    return create(type, button, mouseEvent, pointerId, pointerType, typeCanBubble(type), typeIsCancelable(type));
+}
+
+Ref<PointerEvent> PointerEvent::create(const AtomString& type, MouseButton button, const MouseEvent& mouseEvent, PointerID pointerId, const String& pointerType, CanBubble canBubble, IsCancelable isCancelable)
+{
+    return adoptRef(*new PointerEvent(type, button, mouseEvent, pointerId, pointerType, canBubble, isCancelable));
 }
```
Comment (nt1m) [inline Source/WebCore/dom/PointerEvent.cpp:77]: These changes (while reasonable) seem unrelated to the `getCoalescedEvents` implementation. Can you add an explanation in the commit message?

**Review by nt1m — APPROVED** [review]

File: Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml (hunk 1 above)
Comment (rr-codes) [reply Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:2834]: I plan on making it `testable` once I have an actual implementation that does something
